src/service/session.ts

```diff
@@ -0,0 +1,97 @@
+import { App } from '../entry/app';
+import http from 'http';
+import * as Cookie from 'cookie';
+import crypto from 'crypto';
+import { RouterHandlerResult } from '../router';
+
+function expireSeconds(expireAt:Date) : number {
+    return Math.floor((expireAt.getTime() - Date.now()) / 1000);
+}
+export class SessionService {
+    private cookieName = 'backend';
+    constructor(private app:App) {
+    }
+    private sessionKey = (session:string) => {
+        return `backend:user:session:${session}`;
+    };
+    private sha256Encode = (data:string, salt:string) => {
```

Comment:
Two questions:

- why is salt needed
- why use the same salt for all sessions

Reply:
Theoretically its a wrong design, attackers can always generate enough rainbow tables offline to break things.

A fixed salt can only help a little by blocking newbiee hackers. But yes, it's wrong.

Not using bcrypt because it's slow, and cookie is verified too often. i only use bcrypt for passwords.

Some user-related salt can help. (salts updated in cron job).